server/dialect-idms/src/main/java/org/eclipse/lsp/cobol/dialects/idms/IdmsDialectVisitor.java

```diff
@@ -103,6 +105,37 @@ public List<IdmsCopybookDescriptor> visitStartRule(IdmsParser.StartRuleContext c
     return result;
   }
 
+  @Override
+  public List<IdmsCopybookDescriptor> visitIdmsOnClause(IdmsParser.IdmsOnClauseContext ctx) {
+    Optional.ofNullable(ctx)
+            .map(IdmsParser.IdmsOnClauseContext::generalIdentifier)
+            .map(VisitorHelper::getName)
+            .ifPresent(identifier -> {
+
+              Location location =
+                      context.getExtendedDocument().mapLocation(DialectUtils.constructRange(ctx.generalIdentifier()));
+
+              String generatedText = generateIfStatement(location.getRange().getStart().getCharacter(), identifier);
+              context.getExtendedDocument().replace(location.getRange(), generatedText);
+            });
+
+    return visitChildren(ctx);
+  }
+
+  private String generateIfStatement(int character, String identifier) {
+    final StringBuilder builder = new StringBuilder();
+    builder.append(identifier);
```

Comment:
Why do you echo the identifier?

Reply:
Because it was replaced, but if it is not clear it means it should be updated

Reply:
I think there should be just spaces to move the text into the area b.